### ISSUE TITLE
Should the engine sanitize formspec fields?

### ISSUE THREAD

**Issue by sfan5** [minetest/minetest#14875]
we did something in MTG here: https://github.com/minetest/minetest_game/pull/3022

After copying this almost verbatim into my game I wondered if the engine should perhaps do this.
By "this" I mean stripping control characters. Length limits should be up to mods but I don't see a single reason why formspec fields need to contain zero bytes or the various control characters.

**Comment by wsor4035** [comment]
my two cents: should be optional, in web dev generally your raw post/get etc is always accessible, but you should filter the input first. additionally filters can introduce new bugs sometimes(*glances at minetest history of anticheat and being buggy*)

**Comment by appgurueu** [comment]
I agree with sfan that most probably the engine should do this. Control characters simply aren't valid input here. I don't think Irrlicht's input fields support them properly (feel free to correct me if I'm wrong). Hence, if a client is sending them, we know that this isn't a legitimate player interaction, and we should ignore it appropriately (calling the on cheat callbacks, not calling on receive fields callbacks).

I don't see how this could introduce bugs either; I'm yet to see a legitimate use case being broken by this.

Most properly the opposite is the case: This fixes many potential bugs, perhaps even vulnerabilities, in mod code which does not do this sanitization. Hence it is important for this *not* to be optional so mods can *rely* on formspec fields being sanitized.

(Ideally we should perhaps even do sanitization based on the formspec - so for example inside a field newlines aren't allowed, in a text area they are - but that would be a higher hanging fruit.)

I don't think the analogy with POST/GET requests really holds. These requests are designed to support "raw" data. Minetest formspec input is not designed to support raw data.
